File: algorithms/Flip Columns For Maximum Number of Equal Rows/Flip Columns For Maximum Number of Equal Rows.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        int ans = 0;
        for(int i = 0; i < m; i++){
            vector<int> temp(n);
            for(int j = 0; j < n; j++)
            temp[j] = !matrix[i][j];
            int count = 0;
            for(int j = i; j < m; j++){       
                
                if((matrix[i] == matrix[j]) || (temp == matrix[j]))
                    count++;
            }
            ans = max(ans, count);
        }
        return ans;
    }
};

File: scripts/algorithms/F/Flip Columns For Maximum Number of Equal Rows/Flip Columns For Maximum Number of Equal Rows.py
from typing import List
from collections import Counter


class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        counter = Counter()
        max_val = 2 ** len(matrix[0]) - 1
        for row in matrix:
            v = self.calculate_binary(row)
            counter[v] += 1
            counter[max_val - v] += 1
            # print(f'counter: {counter}')

        return counter.most_common(1)[0][1]

    def calculate_binary(self, row: List[int]) -> int:
        val = 0
        for r in row:
            val = val * 2 + r
        return val
